### ISSUE TITLE
(critical) pbs/torque scheduler jobs do not run

### ISSUE THREAD

**Issue by broeder-j** [marvel-nccr/marvel-virtualmachine#20]
If I submit a job to the scheduler on the vm (through the daemon or manuel) it stays in the queue and does NOT run. (therefore testing on the machine was not possible so far)
My guess is that pbs/torque is not running right.

**Comment by broeder-j** [comment]
Or if the user first has to execute something else, it should be explained somewhere

**Comment by ltalirz** [comment]
@giovannipizzi This is important, could you please have a look?

```
vagrant@marvelvm:~$ qsub run.sh 
0.marvelvm
vagrant@marvelvm:~$ qstat
Job id                    Name             User            Time Use S Queue
------------------------- ---------------- --------------- -------- - -----
0.marvelvm                run.sh           vagrant                0 Q batch     
```

**Comment by giovannipizzi** [comment]
Did the hostname or the IP of the machine change? It could still be about the qmgr, if the server is happy by the scheduler or the mom aren't

**Comment by ltalirz** [comment]
I don't think it changed
```
(aiida) vagrant@marvelvm:~/.local/share/marvelnccr$ ifconfig 
enp0s3    Link encap:Ethernet  HWaddr 08:00:27:a8:df:8b  
          inet addr:10.0.2.15  Bcast:10.0.2.255  Mask:255.255.255.0
          inet6 addr: fe80::a00:27ff:fea8:df8b/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:35605 errors:0 dropped:0 overruns:0 frame:0
          TX packets:17214 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:37309631 (37.3 MB)  TX bytes:2102618 (2.1 MB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:4558 errors:0 dropped:0 overruns:0 frame:0
          TX packets:4558 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:1579927 (1.5 MB)  TX bytes:1579927 (1.5 MB)
```

**Comment by giovannipizzi** [comment]
Is there anything in the logs of the scheduler (server, mom, scheduler)? They should be in /var/spool/... I think.
I will try later to download the VM to see if I can reproduce the problem, but I think the problem should be found in there.

Also: how many CPUs were asked in the script? A common problem might be that you are asking resources that are not available in any queue (e.g. too much memory or too many CPUs)

**Comment by ltalirz** [comment]
my script `run.sh` was
```
#!/bin/bash

cd $PBS_HOME
pw.x -i dft1.in 
```

**Comment by giovannipizzi** [comment]
What is the output of `qstat -f 0` (`0` is the job id)?
Interesting to read it all to see default values, e.g. default walltime, and also in particular I think there is a 'reason' field for queued jobs.

**Comment by ltalirz** [comment]
```
(aiida) vagrant@marvelvm:~$ qstat -f 0
Job Id: 0.marvelvm
    Job_Name = run.sh
    Job_Owner = vagrant@marvelvm
    job_state = Q
    queue = batch
    server = marvelvm
    Checkpoint = u
    ctime = Thu Nov  2 15:37:58 2017
    Error_Path = marvelvm:/home/vagrant/run.sh.e0
    Hold_Types = n
    Join_Path = n
    Keep_Files = n
    Mail_Points = a
    mtime = Thu Nov  2 15:37:58 2017
    Output_Path = marvelvm:/home/vagrant/run.sh.o0
    Priority = 0
    qtime = Thu Nov  2 15:37:58 2017
    Rerunable = True
    Resource_List.nodect = 1
    Resource_List.nodes = 1
    Resource_List.walltime = 01:00:00
    Variable_List = PBS_O_QUEUE=batch,PBS_O_HOST=marvelvm,
	PBS_O_HOME=/home/vagrant,PBS_O_LANG=en_US.UTF-8,PBS_O_LOGNAME=vagrant,
	PBS_O_PATH=/home/vagrant/bin:/home/vagrant/.local/bin:/usr/local/sbin
	:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/ga
	mes:/snap/bin:/home/vagrant/codes/qe-6.1/bin/:/home/vagrant/codes/yamb
	o-4.2.0/bin/:/home/vagrant/codes/fleur/build/:/home/vagrant/codes/sies
	ta-4.0.1/Obj/:/home/vagrant/codes/cp2k-5.1/exe/,PBS_O_SHELL=/bin/bash,
	PBS_SERVER=marvelvm,PBS_O_WORKDIR=/home/vagrant
    etime = Thu Nov  2 15:37:58 2017
    submit_args = run.sh
    fault_tolerant = False
```

**Comment by giovannipizzi** [comment]
ok, then I think we really need to look in the `/var/spool/torque/*_logs/*` files...

**Comment by ltalirz** [comment]
So, the problem is in `/var/spool/torque/server_logs/20171102`:
```
11/02/2017 15:32:40;0086;PBS_Server;Svr;PBS_Server;Recovered queue batch
11/02/2017 15:32:40;0002;PBS_Server;Svr;PBS_Server;Expected 1, recovered 1 queues
11/02/2017 15:32:40;0002;PBS_Server;Svr;PBS_Server;Expected 0, recovered 0 jobs
11/02/2017 15:32:40;0006;PBS_Server;Svr;PBS_Server;Using ports Server:15001  Scheduler:15004  MOM:15002 (server: 'marvelvm')
11/02/2017 15:32:40;0002;PBS_Server;Svr;daemonize_server;INFO:      parent is exiting
11/02/2017 15:32:40;0002;PBS_Server;Svr;daemonize_server;INFO:      parent is exiting
11/02/2017 15:32:40;0002;PBS_Server;Svr;daemonize_server;INFO:      child process in background
11/02/2017 15:32:40;0002;PBS_Server;Svr;PBS_Server;Server Ready, pid = 1533, loglevel=0
11/02/2017 15:32:40;0004;PBS_Server;Svr;WARNING;ALERT: unable to contact node marvelvm
11/02/2017 15:32:45;0040;PBS_Server;Req;ping_nodes;ping attempting to contact 1 nodes
11/02/2017 15:32:45;0002;PBS_Server;Svr;PBS_Server;Torque Server Version = 2.4.16, loglevel = 0
11/02/2017 15:37:45;0002;PBS_Server;Svr;PBS_Server;Torque Server Version = 2.4.16, loglevel = 0
11/02/2017 15:37:58;0100;PBS_Server;Job;0.marvelvm;enqueuing into batch, state 1 hop 1
11/02/2017 15:37:58;0008;PBS_Server;Job;0.marvelvm;Job Queued at request of vagrant@marvelvm, owner = vagrant@marvelvm, job name = run.sh, queue = batch
```
Can you please have a look?
There are a number of other things to fix and I have never set up torque by myself.

**Comment by giovannipizzi** [comment]
I will, I'll check first if I can reproduce the problem

**Comment by giovannipizzi** [comment]
So, I think I understood the problem and I'll briefly describe it here. I'll try to commit a fix ASAP, after a few final checks.

- the problem is the `pbs_sched` process is dead. In the logs, we find: `pbs_sched Cannot assign requested address (99) in main, bind`. The reason is that the startup of the service happens too early, before a DHCP address is set.
- manually restarting the service works (`/etc/init.d/torque-scheduler restart`). However to make everything work again, one needs (at least in my experience) also to restart the `torque-server`, and then to submit any calculation to "unlock" also the old ones from the QUEUED state
- I prepared a script to be put in the `if-up.d` that will also solve #4 to rewrite `/etc/hosts/` with the correct IP address, and to restart the `scheduler` if it is down but the server is up
- Unfortunately, this is not enough: the scheduler starts anyway too early, and when the `if-up` script is run, the server is still down (the server instead waits for the network) and therefore there is no solution
- However, I found a "bug" in the headers of the `/etc/init.d/torque-scheduler` script, that didn't put a dependency on the network being up; fixing this should be enough to solve the problem.

I'm working on a commit with these fixes.
